regolith/helpers/a_proposalhelper.py

```diff
@@ -0,0 +1,197 @@
+                       )
+    subpi.add_argument("duration", help="Duration of proposal in years"
+                       )
+    subpi.add_argument("title", help="Actual title of the proposal"
+                       )
+    subpi.add_argument("-c", "--currency",
+                       help="Currency in which amount is specified. Defaults to USD"
+                       )
+    subpi.add_argument("-p", "--pi",
+                       help="Name of principal investigator. Defaults to"
+                            "group pi name in database"
+                       )
+    subpi.add_argument("--cppflag", help="Current and pending form (true or false)"
+                       )
+    subpi.add_argument("--other_agencies", help="Other agencies to which the proposal has been "
+                                                "submitted"
+                       )
+    subpi.add_argument("--institution", help="The institution where the work will primarily"
+                                             "be carried out"
+                       )
+    subpi.add_argument("--months_academic", help="Number of working months in the academic year"
+                                                 "to be stated on the current and pending form"
+                       )
+    subpi.add_argument("--months_summer", help="Number of working months in the summer to be"
+                                                "stated on the current and pending form"
+                       )
+    subpi.add_argument("--scope", help="Scope of the project"
+                       )
+    subpi.add_argument("-f", "--funder",
+                       help="Agency where the proposal is being submitted"
+                       )
+    subpi.add_argument("-n", "--notes", nargs="+",
+                       help="Anything to note"
+                       )
+    return subpi
+
+
+class ProposalAdderHelper(DbHelperBase):
+    """Helper for adding a proposal to the proposals collection.
+
+       A proposal is a dictionary object describing a research or
+        project proposal submitted by the group.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "a_proposal"
+    needed_dbs = [f'{TARGET_COLL}', 'people']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        gtx = self.gtx
+        rc = self.rc
+        if not rc.date:
```

Comment:
we are not asking for date in the CLI so just use today's date.

Reply:
will do – would you prefer that the suffix to the key, for example, `20_proposal`, be the year from the `begin_date` or `now.year` ?